src/utils/utils.ts

```diff
@@ -141,6 +144,10 @@ export function primitiveTypeToCairo(
 
   if (uintMatch) {
     const bits = BigInt(uintMatch[1]);
+    if (bits === 252n) {
```

Comment:
Yes felt is lesser than 252 bits, but bigger than 251. The word size in Cairo is actually 252 bits.

The reason for this change is that previously (old warp) we could easily represent felts in the transpiled contract by using a any uint lesser than 256. 

Now in Cairo 1 there is an intention of having a `ux` per `uintx`, which means we have no way of representing `felts` now in the transpiled contracts. 

The use of `252n` is a hacky way of the transpiler to understand that we mean this type is going to be a felt. We cannot create our own `felt252` typestring because the solc typed ast which we use to parse all types everywhere would not know where it come from and error.

We decided this way because it was the much lesser evil among other options.  Also is easy to revert back if we need it.

Reply:
Oh, does solc understand uint252? I'm surprised given it's not a legal Solidity type, it's not divisible by 8. If that's the case though, feel free to resolve

Reply:
Yes, luckily for us at this time